aten/src/ATen/native/cpu/MaxPooling.cpp

```diff
@@ -0,0 +1,57 @@
+#include <ATen/ATen.h>
+#include <ATen/Parallel.h>
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/MaxPooling.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+template <typename scalar_t>
+inline void max_pool1d_kernel(
+    scalar_t* C10_RESTRICT op,
+    const scalar_t* C10_RESTRICT ip,
+    const PoolingParams1D& p) {
+  for (int64_t kj = 0; kj < p.KW; ++kj) {
+    int64_t oj = p.valid_output_start(kj);
+    int64_t oe = p.valid_output_end(kj);
+    int64_t ij = p.index(kj, oj);
+    for (; oj < oe; ++oj, ij += p.SJ) {
+      scalar_t val = ip[ij];
+      bool update_max = std::isnan(val) || op[oj] < val;
```

Comment:
way to blow the cache - instead of computing max in registers within a window, you are constantly accessing full output line.

Reply:
True, and I have plans for a future PR to compute output in blocks that fit within the cache. For now I wanted to keep the code simple so I can have confidence it works. This way the compiler auto vectorized the loop for me.